### PR TITLE
[protobuf-c] Update to 1.4.0

### PR DESCRIPTION
**Pull request by chausner**
**Describe the pull request**

- #### What does your PR fix?  
  Updates protobuf-c to 1.4.0

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All, Yes. All features have been tested on all Windows triplets.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run ./vcpkg x-add-version --all and committed the result?  
  Yes

### CODE REVIEW

File: ports/protobuf-c/vcpkg.json [hunk 1]
```diff
@@ -1,15 +1,18 @@
 {
   "name": "protobuf-c",
-  "version-string": "1.3.2",
-  "port-version": 3,
+  "version-semver": "1.4.0",
```
Comment (chausner) [inline ports/protobuf-c/vcpkg.json:3]: Adheres to semver according to https://github.com/protobuf-c/protobuf-c#versioning.

File: ports/protobuf-c/fix-crt-linkage.patch [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+-	foreach(flag_var
+-		CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
+-		CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO
+-		CMAKE_C_FLAGS CMAKE_C_FLAGS_DEBUG CMAKE_C_FLAGS_RELEASE 
+-		CMAKE_C_FLAGS_MINSIZEREL CMAKE_FLAGS_RELWITHDEBINFO)
+-	  if(${flag_var} MATCHES "/MD")
+-		string(REGEX REPLACE "/MD" "/MT" ${flag_var} "${${flag_var}}")
+-	  endif(${flag_var} MATCHES "/MD")
+-	endforeach(flag_var)
+-endif (MSVC AND NOT BUILD_SHARED_LIBS)
```
Comment (NancyLi1013) [inline ports/protobuf-c/fix-crt-linkage.patch:23]: Could you please use `if(0)` to minimize the patch? Please see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#minimize-patches.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/protobuf-c/fix-crt-linkage.patch (hunk 2 above)
Comment (chausner) [reply ports/protobuf-c/fix-crt-linkage.patch]: Thanks for the hint! Done in dbd88961a035da115b6457fbe58d81968026f659.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM, thanks for your updates @chausner.

**Pull request merged**